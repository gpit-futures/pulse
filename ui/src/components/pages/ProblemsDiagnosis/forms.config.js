export const valuesNames = {
  PROBLEM: 'problem',
  DATE_OF_ONSET: 'dateOfOnset',
  DESCRIPTION: 'description',
  NHS_WEB_PAGE_URL: 'nhsURL',
  TERMINOLOGY: 'terminology',
  CODE: 'code',
  AUTHOR: 'author',
  DATE: 'date',
  DATE_CREATED: 'dateCreated',

  IMPORT: 'originalSource',
  IS_IMPORT: 'isImport',
  ORIGINAL_SOURCE: 'originalSource',
  ORIGINAL_COMPOSITION: 'originalComposition',

  SOURCE: 'source',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  PROBLEM: 'Problem / Diagnosis',
  DATE_OF_ONSET: 'Date of Onset',
  DESCRIPTION: 'Description',
  NHS_WEB_PAGE_URL: 'NHS Choices Web Page URL',
  TERMINOLOGY: 'Terminology',
  CODE: 'Terminology Code',
  AUTHOR: 'Author',
  DATE: 'Date',

  IMPORT: 'Import URL',
  IS_IMPORT: 'Is Imported',
  ORIGINAL_SOURCE: 'Original Source',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};
