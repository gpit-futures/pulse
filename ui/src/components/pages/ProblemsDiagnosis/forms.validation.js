import { valuesNames } from './forms.config';

const validateForm = (values) => {
  const errors = {};
  errors[valuesNames.PROBLEM] = !values[valuesNames.PROBLEM] ? 'You must enter a value.' : null;
  errors[valuesNames.DESCRIPTION] = !values[valuesNames.DESCRIPTION] ? 'You must enter a value.' : null;
  errors[valuesNames.DATE_OF_ONSET] = !values[valuesNames.DATE_OF_ONSET] ? 'You must enter a valid date.' : null;
  return errors
};

export { validateForm }
