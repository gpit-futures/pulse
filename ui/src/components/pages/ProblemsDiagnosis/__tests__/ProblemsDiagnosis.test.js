import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import ProblemsDiagnosis from '../ProblemsDiagnosis';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsDiagnoses: {
    '9999999024': [
      {
        [valuesNames.PROBLEM]: 'Cholecystectomy',
        [valuesNames.DATE_OF_ONSET]: 1491861600000,
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: '0318e94e-1803-46fe-a92a-647eca029323',
        dateOfOnsetConvert: '11-Apr-2017',
      },
    ],
  },
};
const storeWithFormsError = mockStore(Object.assign({
  diagnosesDetail: {
    '9999999000': {
      [valuesNames.PROBLEM]: '1.0',
      [valuesNames.DATE_OF_ONSET]: 1510272000000,
      [valuesNames.DESCRIPTION]: 'sdsda',
      [valuesNames.TERMINOLOGY]: 'dadadyuio',
      [valuesNames.CODE]: '1.239389E7',
      [valuesNames.AUTHOR]: 'Dr Tony Shannon',
      [valuesNames.DATE_CREATED]: 1511284969000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: '8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2',
    },
  },
  form: {
    diagnosesPanelFormSelector: {
      syncErrors: {
        [valuesNames.PROBLEM]: 'You must enter a value.',
        [valuesNames.DESCRIPTION]: 'You must enter a value.',
      },
    },
    diagnosesCreateFormSelector: {
      syncErrors: {
        [valuesNames.PROBLEM]: 'You must enter a value.',
        [valuesNames.DESCRIPTION]: 'You must enter a value.',
      },
    },
  },
}, storeResource));
const storeEmpty = mockStore(Object.assign({
  diagnosesDetail: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  diagnosesDetail: {
    '9999999000': {
      [valuesNames.PROBLEM]: '1.0',
      [valuesNames.DATE_OF_ONSET]: 1510272000000,
      [valuesNames.DESCRIPTION]: 'sdsda',
      [valuesNames.TERMINOLOGY]: 'dadadyuio',
      [valuesNames.CODE]: '1.239389E7',
      [valuesNames.AUTHOR]: 'Dr Tony Shannon',
      [valuesNames.DATE_CREATED]: 1511284969000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: '8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2',
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/diagnoses`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextCreate = generateNewContext(context, `/patients/${userId}/diagnoses/create`);
const contextDetail = generateNewContext(context, `/patients/${userId}/diagnoses/${sourceId}`);

const contextImport = {
  router: {
    route: { match: { params: { userId } } },
    history: {
      push: () => {},
      replace: () => {},
      goBack: () => {},
      location: {
        pathname: `/patients/${userId}/diagnoses/create`,
        state: {
          importData: {
            isImport: true,
            originalSource: 'domen.com/documents/documents_id',
            name: 'name',
          }
        }
      },
    }
  },
};

// configuration of forms for testing methods
const formValuesEdit = {
  [valuesNames.PROBLEM]: '1.0e',
  [valuesNames.DATE_OF_ONSET]: 1510272000000,
  [valuesNames.DESCRIPTION]: 'sdsda',
  [valuesNames.TERMINOLOGY]: 'dadadyuio',
  [valuesNames.CODE]: '1.239389E7',
  [valuesNames.AUTHOR]: 'Dr Tony Shannon',
  [valuesNames.IS_IMPORT]: true,
};
const formValuesCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.SOURCE_ID]: '',
  [valuesNames.IS_IMPORT]: false,
  [valuesNames.CODE]: '12393890',
  [valuesNames.TERMINOLOGY]: 'SNOMED-CT',
  [valuesNames.PROBLEM]: 'test',
  [valuesNames.DESCRIPTION]: 'test',
  [valuesNames.DATE_OF_ONSET]: '20-Dec-2017',
};
const formValuesImportCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.SOURCE_ID]: '',
  [valuesNames.IS_IMPORT]: true,
  [valuesNames.IMPORT]: 'domen.com/documents/documents_id',
  [valuesNames.CODE]: '12393890',
  [valuesNames.TERMINOLOGY]: 'SNOMED-CT',
  [valuesNames.PROBLEM]: 'test',
  [valuesNames.DESCRIPTION]: 'test',
  [valuesNames.DATE_OF_ONSET]: '20-Dec-2017',
};
const match = {
  params: {
    userId,
  },
};

describe('Component <ProblemsDiagnosis />', () => {
  it('should renders correctly with diagnosisDetail and testing Detail Panel', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    // Testing component handleDetailDiagnosesClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ProblemsDiagnosisDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    component.instance().handleDetailDiagnosesClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isBtnCreateVisible: true, isCreatePanelVisible: false, openedPanel: 'diagnosesPanel', editedPanel: {}, expandedPanel: 'all', isLoading: true });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('ProblemsDiagnosisDetail')).toHaveLength(1);
    expect(component.find('PluginCreate')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('diagnosesPanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('diagnosesPanel', 'diagnosesDetail');
    expect(component.state().openedPanel).toEqual('diagnosesPanel');
    expect(component.state().expandedPanel).toEqual('diagnosesPanel');
    component.setState({ openedPanel: 'diagnosesPanel', expandedPanel: 'diagnosesPanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    component.setState({ openedPanel: 'diagnosesPanel', expandedPanel: 'all' });

    component.instance().handleExpand('diagnosesPanel', 'diagnosesCreate');
    component.setState({ expandedPanel: 'all' });
    component.instance().handleExpand('diagnosesPanel', 'diagnosesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('diagnosesPanel', 'diagnosesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('diagnosesPanel', 'diagnosesCreate');

    // Testing component detail form methods
    expect(component.state().editedPanel).toEqual({});
    component.instance().handleEdit('diagnosesPanel');
    expect(component.state().editedPanel).toEqual({ diagnosesPanel: true });
    component.instance().handleDiagnosisDetailCancel('diagnosesPanel');
    expect(component.state().editedPanel).toEqual({ diagnosesPanel: false });
    component.instance().handleEdit('diagnosesPanel');
    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'diagnosesPanel');
    expect(component.state().editedPanel).toEqual({ diagnosesPanel: false });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with diagnosesDetail and testing Create Panel', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ProblemsDiagnosisDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    // Testing component create panel methods
    component.instance().handleCreate();
    const componentStateAfterMethod = component.state();
    component.setState({ isBtnCreateVisible: false, isCreatePanelVisible: true, openedPanel: 'diagnosesCreate', isSecondPanel: true, isDetailPanelVisible: false, isBtnExpandVisible: true, expandedPanel: 'all', isSubmit: false, isLoading: true })
    const componentStateAfterSetState = component.state();
    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('PluginCreate')).toHaveLength(1);
    expect(component).toMatchSnapshot();
    component.instance().handleCreateCancel();
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    // Testing component hideCreateForm methods
    component.instance().hideCreateForm();
    expect(component.state().isBtnCreateVisible).toEqual(true);
    expect(component.state().isCreatePanelVisible).toEqual(false);
    expect(component.state().openedPanel).toEqual('diagnosesPanel');
    expect(component.state().isSecondPanel).toEqual(false);
    expect(component.state().expandedPanel).toEqual('all');
    expect(component.state().isBtnExpandVisible).toEqual(false);

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual('problem');
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'dateOfOnset', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('dateOfOnset');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextCreate);
    component.setProps({ test: 'testing create context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with forms error', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeWithFormsError}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'diagnosesPanel');
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly when data take from Documents how "import"', () => {
    const component = shallow(
      <ProblemsDiagnosis
        store={storeEmpty}
        match={match}
      />, { context: contextImport }).dive().dive().dive().dive().dive().dive();

    component.instance().goBack();
    component.instance().handleSaveSettingsCreateForm(formValuesImportCreate);

    expect(component).toMatchSnapshot();
  });
});
