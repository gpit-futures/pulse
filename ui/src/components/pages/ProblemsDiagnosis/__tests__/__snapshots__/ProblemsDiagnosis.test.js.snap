// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ProblemsDiagnosis /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="problems.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Problems / Diagnoses"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly and testing context and lifecycle componentWillReceiveProps 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            id="8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ProblemsDiagnosisDetail
          currentPanel="diagnosesDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "code": "1.239389E7",
              "dateCreated": 1511284969000,
              "dateOfOnset": 1510272000000,
              "description": "sdsda",
              "problem": "1.0",
              "source": "ethercis",
              "sourceId": "8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2",
              "terminology": "dadadyuio",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="diagnosesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diagnosesCreate"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly when data take from Documents how "import" 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="problems.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Problems / Diagnoses"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with diagnosesDetail and testing Create Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              isSubmit={false}
            />
          }
          currentPanel="diagnosesCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="diagnosesCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onGoBack={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diagnosesCreate"
          title="Create Problem and Diagnosis"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with diagnosesDetail and testing Create Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              isSubmit={false}
            />
          }
          currentPanel="diagnosesCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="diagnosesCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onGoBack={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diagnosesCreate"
          title="Create Problem and Diagnosis"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with diagnosisDetail and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            id="8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ProblemsDiagnosisDetail
          currentPanel="diagnosesDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "code": "1.239389E7",
              "dateCreated": 1511284969000,
              "dateOfOnset": 1510272000000,
              "description": "sdsda",
              "problem": "1.0",
              "source": "ethercis",
              "sourceId": "8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2",
              "terminology": "dadadyuio",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="diagnosesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diagnosesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with diagnosisDetail and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ProblemsDiagnosisDetail
          currentPanel="diagnosesDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "code": "1.239389E7",
              "dateCreated": 1511284969000,
              "dateOfOnset": 1510272000000,
              "description": "sdsda",
              "problem": "1.0",
              "source": "ethercis",
              "sourceId": "8abeb5bb-717e-4b25-9bbf-d79f5e7a34e2",
              "terminology": "dadadyuio",
            }
          }
          editedPanel={Object {}}
          expandedPanel="diagnosesPanel"
          isSubmit={false}
          name="diagnosesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="diagnosesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with diagnosisDetail and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    />
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="problems.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Problems / Diagnoses"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <ProblemsDiagnosis /> should renders correctly with forms error 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="problems.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Problems / Diagnoses"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="diagnosesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="diagnosesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Problems / Diagnoses"
          />
          <PluginMainPanel
            columnNameSortBy="problem"
            emptyDataMessage="No diagnoses"
            headers={
              Array [
                Object {
                  "key": "problem",
                  "title": "Problems / Diagnoses",
                  "width": "56%",
                },
                Object {
                  "key": "dateOfOnset",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="diagnoses"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;
